tradeexecutor/analysis/trade_analyser.py

```diff
@@ -451,14 +474,34 @@ def calculate_summary_statistics(self) -> TradeSummary:
 
             if position.is_win():
                 won += 1
+                average_winning_trade.append(position.realised_profit_percent)
+                average_winning_trade_duration.append(position.duration)
+
             elif position.is_lose():
                 lost += 1
+                average_losing_trade.append(position.realised_profit_percent)
+                average_losing_trade_duration.append(position.duration)
+
             else:
                 # Any profit exactly balances out loss in slippage and commission
                 zero_loss += 1
 
             profit += position.realised_profit
 
+        average_winning_trade_profit_pc = np.mean(average_winning_trade)
+        average_losing_trade_loss_pc = np.mean(average_losing_trade)
+        if average_winning_trade:
+            biggest_winning_trade_pc = max(average_winning_trade)
+        else:
+            biggest_winning_trade_pc = np.nan
```

Comment:
Would it make more sense to just use `None` for unavailable stats?

Reply:
We are returning the data to the notebook, not sure if `None` will be picked by the notebook. For notebook purposes that is why I am using `np.nan`

Reply:
@m-root this isn't a correct assumption, the stats are used to displayed in notebook and will also be stored in state file (JSON) so you should find a way to make it suitable for both

Reply:
Fair point, let;s go with `None` then instead of a specific object from a library. `None` is so ubiquitous that "every" part of the ecosystem (libraries, renderers, serializers...) must know how to handle it.